toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -303,13 +304,16 @@ func testSRPMFile(agent buildagents.BuildAgent, checkAttempts int, basePackageNa
 			return
 		}
 
-		buildErr = parseCheckSection(logFile)
-		checkFailed = (buildErr != nil)
+		checkFailed, buildErr = parseCheckSection(logFile)
+		// If the build succeeded but tests failed, we still want to retry.
+		if buildErr == nil && checkFailed {
+			buildErr = fmt.Errorf("package test for '%s' failed", basePackageName)
+		}
 		return
 	}, checkAttempts, retryDuration)
 
-	if err != nil && checkFailed {
-		logger.Log.Warnf("Tests failed for '%s'. Error: %s", srpmFile, err)
+	if checkFailed {
+		logger.Log.Debugf("Tests failed for '%s' after %d retries.", basePackageName, checkAttempts)
```

Comment:
why the demotion from warning to debug? this seems worthy of a warning

Reply:
Good eye. The info level message is now inside `PrintBuildResult`, because I've noticed that's where the rest of build/test logs come from. This one here is just an additional bit with the number of retries, which are internal to the tooling, so the user probably doesn't need to see that all the time when a test fails.